tsl/src/bgw_policy/job_api.c

```diff
@@ -106,11 +115,16 @@ Datum
 job_delete(PG_FUNCTION_ARGS)
 {
 	int32 job_id = PG_GETARG_INT32(0);
+	BgwJob *job;
+	Oid owner;
 
 	TS_PREVENT_FUNC_IF_READ_ONLY();
 
-	BgwJob *job = ts_bgw_job_find(job_id, CurrentMemoryContext, true);
-	Oid owner = get_role_oid(NameStr(job->fd.owner), false);
+	if (PG_ARGISNULL(0))
+		ereport(ERROR, (errcode(ERRCODE_INVALID_PARAMETER_VALUE), errmsg("job ID cannot be NULL")));
+
+	job = ts_bgw_job_find(job_id, CurrentMemoryContext, true);
```

Comment:
Checking job ID argument and retrieving the job is the same code in all job functions, except `add_job`. It might be beneficial to replace code copies with a call to a function or macro.